src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-        """
-        Constructor requiring the server address, the user and password of DSP
-        :param server: Address of the server, e.g https://api.dasch.swiss
-        :param prefixes: Ontology prefixes used
-        """
-
-        self._server = regex.sub(r"\/$", "", server)
-        self._prefixes = prefixes
-        self._token = None
-        self._log = False
+    server: str
+    prefixes: Optional[dict[str, str]] = None
+    token: Optional[str] = None
+    log: bool = False
+    log_directory: Path = Path("HTTP requests")
 
     def login(self, email: str, password: str) -> None:
         """
-        Method to login into DSP which creates a session token.
-        :param email: Email of user, e.g., root@example.com
-        :param password: Password of the user, e.g. test
-        """
-
-        credentials = {"email": email, "password": password}
-        jsondata = json.dumps(credentials)
+        Retrieve a session token.
 
+        Args:
+            email: Email of user
+            password: Password of the user
+        """
         response = requests.post(
-            self._server + "/v2/authentication",
+            self.server + "/v2/authentication",
             headers={"Content-Type": "application/json; charset=UTF-8"},
-            data=jsondata,
+            data=json.dumps({"email": email, "password": password}),
             timeout=20,
         )
         check_for_api_error(response)
         result = response.json()
-        self._token = result["token"]
-
-    def get_token(self) -> str:
-        """
-        Returns the token
-        :return: token string
-        """
-
-        return self._token
-
-    @property
-    def token(self) -> str:
-        """Get the token of this connection"""
-        return self._token
+        self.token = result["token"]
 
     def start_logging(self) -> None:
         """Start writing every API call to a file"""
-        self._log = True
+        self.log = True
+        self.log_directory.mkdir(exist_ok=True)
 
-    def stop_logging(self):
+    def stop_logging(self) -> None:
         """Stop writing every API call to a file"""
-        self._log = False
+        self.log = False
 
     def logout(self) -> None:
-        """
-        Performs a logout
-        :return: None
-        """
-
-        if self._token is not None:
+        """Delete the session token"""
+        if self.token:
             response = requests.delete(
-                self._server + "/v2/authentication",
-                headers={"Authorization": "Bearer " + self._token},
+                self.server + "/v2/authentication",
+                headers={"Authorization": "Bearer " + self.token},
                 timeout=20,
             )
             check_for_api_error(response)
-            self._token = None
-
-    def __del__(self):
-        pass
-        # self.logout()
-
-    def post(self, path: str, jsondata: Optional[str] = None):
+            self.token = None
+
+    def write_log_file(
```

Comment:
I think I mentioned this before: to me the term "log" suggests something narrower than just writing information to a file, namely a certain line-by-line format with timestamp, loglevel, etc.  
How about `write_http_request`?

Reply:
done